pandas/_libs/lib.pyx

```diff
@@ -1961,7 +1945,7 @@ cdef class BytesValidator(Validator):
         return isinstance(value, bytes)
 
     cdef bint is_array_typed(self) except -1:
-        return issubclass(self.dtype.type, np.bytes_)
+        return self.dtype.type == cnp.NPY_BYTE or self.dtype.type == cnp.NPY_UBYTE
```

Comment:
cc @seberg is this the right way to check for bytes?

Reply:
No, its `NPY_STRING` (string is `NPY_UNICODE`, leftover from python 2)